Library/Homebrew/linkage_checker.rb

```diff
@@ -168,21 +168,29 @@ def check_dylibs(rebuild_cache:)
     keg_files_dylibs.each do |file, dylibs|
       file_has_any_rpath_dylibs = T.let(false, T::Boolean)
       dylibs.each do |dylib|
+        pathname = Pathname(file)
+        # Resolve some variable install names.
+        expanded_path = pathname.dirname
+        dylib.sub!("@loader_path", expanded_path.to_s)
+        dylib.sub!("@executable_path", expanded_path.to_s) if pathname.binary_executable?
+        # Remove redundant dots from the path.
+        dylib = Pathname(dylib).cleanpath.to_s
+
         @reverse_links[dylib] << file
 
         # Files that link @rpath-prefixed dylibs must include at
         # least one rpath in order to resolve it.
         if !file_has_any_rpath_dylibs && (dylib.start_with? "@rpath/")
           file_has_any_rpath_dylibs = true
-          pathname = Pathname(file)
           @files_missing_rpaths << file if pathname.rpaths.empty?
         end
 
         next if checked_dylibs.include? dylib
 
         checked_dylibs << dylib
 
-        if dylib.start_with? "@"
+        if dylib.start_with?("@rpath") ||
+           (dylib.start_with?("@executable_path") && !pathname.binary_executable?)
```

Comment:
This case is a little worrying. While it is technically possible to do this, it means that the linked library (`dylib`) that is loaded depends on the location of the executable that loads `file`. I feel like we should be throwing at least a warning for these with `brew linkage --test --strict`. (But this isn't needed for your PR -- unless you're feeling generous.)

Reply:
Not gonna say it'd wrong with no exceptions (I believe doing that would be fine e.g. within an app bundle that doesn't expect external binaries to link against its libraries), but I don't really expect to see such a thing in Homebrew.

I can agree with a warning + a failure if `--strict` (and I guess I'm feeling generous! 🙂). Done!